### ISSUE TITLE
Identify what breaks when running RSMTool with SKLL2.0

### ISSUE THREAD

**Issue by aloukina** [EducationalTestingService/rsmtool#339]
First step for #311

**Comment by mulhod** [comment]
Updating the version of SKLL to 2.0.0 succeeds in terms of environment creation.

Running the unit tests we encounter a few issues. Here are unit tests errors and failures:

```
======================================================================
ERROR: test_analyzer.TestAnalyzer.test_compute_pca_less_components_than_features
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/tests/test_analyzer.py", line 146, in test_compute_pca_less_components_than_features
    (components, variance) = Analyzer.compute_pca(df, df.columns)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/analyzer.py", line 466, in compute_pca
    pca.fit(df_pca)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/sklearn/decomposition/pca.py", line 341, in fit
    self._fit(X)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/sklearn/decomposition/pca.py", line 407, in _fit
    return self._fit_full(X, n_components)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/sklearn/decomposition/pca.py", line 426, in _fit_full
    % (n_components, min(n_samples, n_features)))
ValueError: n_components=101 must be between 0 and min(n_samples, n_features)=100 with svd_solver='full'

======================================================================
ERROR: test_analyzer.TestAnalyzer.test_that_correlation_helper_works_for_data_with_one_row
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/tests/test_analyzer.py", line 51, in test_that_correlation_helper_works_for_data_with_one_row
    retval = Analyzer.correlation_helper(self.df_features[:1], 'sc1', 'group')
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/analyzer.py", line 551, in correlation_helper
    df_target_cors[group] = df_group.apply(lambda s:
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/frame.py", line 6014, in apply
    return op.get_result()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 318, in get_result
    return super(FrameRowApply, self).get_result()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 142, in get_result
    return self.apply_standard()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 248, in apply_standard
    self.apply_series_generator()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 277, in apply_series_generator
    results[i] = self.f(v)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/analyzer.py", line 552, in <lambda>
    pearsonr(s, df_group[target_variable])[0])
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/scipy/stats/stats.py", line 3501, in pearsonr
    raise ValueError('x and y must have length at least 2.')
ValueError: ('x and y must have length at least 2.', 'occurred at index sc1')

======================================================================
ERROR: test_experiment_rsmtool_1.test_run_experiment_parameterized('lr-subgroups-with-edge-cases', 'lr_subgroups_with_edge_cases', {'subgroups': ['group_edge_cases']})
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/parameterized/parameterized.py", line 387, in <lambda>
    nose_func = wraps(func)(lambda *args: func(*args[:-1], **args[-1]))
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/tests/test_experiment_rsmtool_1.py", line 47, in test_run_experiment_parameterized
    check_run_experiment(*args, **kwargs)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/test_utils.py", line 88, in check_run_experiment
    do_run_experiment(source, experiment_id, config_file)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/test_utils.py", line 330, in do_run_experiment
    run_experiment(config_file, experiment_dir)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/rsmtool.py", line 221, in run_experiment
    processed_config)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/analyzer.py", line 1422, in run_training_analyses
    include_length=include_length)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/analyzer.py", line 846, in compute_correlations_by_group
    include_length=include_length)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/analyzer.py", line 551, in correlation_helper
    df_target_cors[group] = df_group.apply(lambda s:
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/frame.py", line 6014, in apply
    return op.get_result()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 318, in get_result
    return super(FrameRowApply, self).get_result()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 142, in get_result
    return self.apply_standard()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 248, in apply_standard
    self.apply_series_generator()
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/pandas/core/apply.py", line 277, in apply_series_generator
    results[i] = self.f(v)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/rsmtool/analyzer.py", line 552, in <lambda>
    pearsonr(s, df_group[target_variable])[0])
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/scipy/stats/stats.py", line 3501, in pearsonr
    raise ValueError('x and y must have length at least 2.')
ValueError: ('x and y must have length at least 2.', 'occurred at index FEATURE1')
-------------------- >> begin captured stdout << ---------------------
The exact solution is  x = 0

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: test_analyzer.TestAnalyzer.test_that_metrics_helper_works_for_data_with_one_row
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/tests/test_analyzer.py", line 93, in test_that_metrics_helper_works_for_data_with_one_row
    assert_equal(evals.isnull().values.sum(), 4)
AssertionError: 5 != 4

======================================================================
FAIL: test_comparer.TestComparer.test_compute_correlations_between_versions_custom_columns
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/tests/test_comparer.py", line 105, in test_compute_correlations_between_versions_custom_columns
    assert_equal(df_cors.at['feature2', 'old_new'], 1.0)
AssertionError: 0.9999999999999999 != 1.0

======================================================================
FAIL: test_comparer.TestComparer.test_compute_correlations_between_versions_default_columns
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/tests/test_comparer.py", line 82, in test_compute_correlations_between_versions_default_columns
    assert_equal(df_cors.at['feature2', 'old_new'], 1.0)
AssertionError: 0.9999999999999999 != 1.0

======================================================================
FAIL: test_comparer.TestComparer.test_compute_correlations_between_versions_extra_data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool_repo/tests/test_comparer.py", line 141, in test_compute_correlations_between_versions_extra_data
    assert_equal(df_cors.at['feature2', 'old_new'], 1.0)
AssertionError: 0.9999999999999999 != 1.0

----------------------------------------------------------------------
Ran 732 tests in 1553.979s

FAILED (errors=3, failures=4)
```

**Comment by desilinguist** [comment]
That's not too bad at all! Looks like we need to update the API calls to SciPy and Sklearn PCA, fix a warning somewhere, and then update some unit tests.

**Comment by desilinguist** [comment]
Addressed and used to create #349 which was merged.
